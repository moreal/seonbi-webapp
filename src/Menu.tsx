import React from "react";

export const Menu: React.FC = () => {
    return <div style={{
        display: "flex",
        alignItems:"center",
        width: "auto",
        minHeight: "80px",
        borderWidth: 0,
        borderBottomWidth: "3px",
        borderColor: "black",
        borderStyle: "solid",
        margin: 0,
        // paddingTop: "20px",
        // paddingBottom: "0",
        paddingLeft: "40px",
    }}>
        <img src="icon.png" style={{ height: "40px", width: "40px" }}/>
        <span style={{ lineHeight: "40px" }}>Seonbi</span>
    </div>
}
