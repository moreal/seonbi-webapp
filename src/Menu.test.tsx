import React from "react";
import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";
import { Menu } from "./Menu";

let root: HTMLDivElement | null = null;

beforeEach(() => {
    root = document.createElement("div");
    document.body.appendChild(root);
});

afterEach(() => {
    if (root !== null) {
        unmountComponentAtNode(root);
        root.remove();
        root = null;
    }
});

describe("Menu", () => {
    it("snapshot", () => {
        const container: HTMLDivElement = root!;
        act(() => {
            render(<Menu />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it("should render", () => {
        const container: HTMLDivElement = root!;
        act(() => {
            render(<Menu />, container);
        })

        expect(container.querySelector("img")?.src).toBe("icon.png");
        expect(container.textContent).toBe("Seonbi");
    })
})
