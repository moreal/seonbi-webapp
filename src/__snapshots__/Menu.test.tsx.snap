// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot 1`] = `"<div style=\\"display:flex;align-items:center;width:auto;min-height:80px;border-width:0;border-bottom-width:3px;border-color:black;border-style:solid;margin:0;padding-left:40px\\"><img src=\\"image.png\\" style=\\"height:40px;width:40px\\"/><span style=\\"line-height:40px\\">Seonbi</span></div>"`;
